### ISSUE TITLE
Solid: Replace react `htmlFor` prop with `for`

### ISSUE THREAD

**Issue by ben-xD** [chakra-ui/ark#1601]
### Description

I get a warning from SolidJS eslint linter:
```ts
$HOME/repos/talkdash/frontend/src/App.tsx
   77:21  warning  Prefer the `for` prop over the deprecated `htmlFor` prop  solid/no-react-specific-props
  111:21  warning  Prefer the `for` prop over the deprecated `htmlFor` prop  solid/no-react-specific-props
```

However, there are usages of `htmlFor` in the solid project:
```ts
type MenuItemGroupLabelParams = {
    htmlFor: string;
};
```

We should use `for` instead of `htmlFor`. :)

### Link to Reproduction

See code

### Steps to reproduce

See code

### Ark UI Version

0.11.0

### Framework

- [ ] React
- [X] Solid
- [ ] Vue

**Comment by Omikorin** [comment]
Hi, I checked [changelog](https://github.com/solidjs/solid/blob/main/CHANGELOG.md#classname-htmlfor-deprecated) and you are right :)
I think this case is a low priority for now (it is only a deprecation warning, so nearest removal could be in a year minimum).

I've created draft #1604 to work on this later.

**Comment by cschroeter** [comment]
@ben-xD 

Thanks for taking your time to report this issue. A fix will be availabe in 1.0
